### PR TITLE
Edkrepo: Enable Configurable Cache Locations

### PR DESCRIPTION
**Pull request by hrsh25**
Updated Cache command to accept an optional path argument
Signed-off-by: Harsh Vora harsh.vora@intel.com

### CODE REVIEW

File: edkrepo/commands/arguments/cache_args.py [hunk 1]
```diff
@@ -17,3 +17,4 @@
 COMMAND_UPDATE_HELP = 'Update the repo cache for all cached projects.'
 COMMAND_INFO_HELP = 'Display the current cache information.'
 COMMAND_PROJECT_HELP = 'Project to add to the cache.'
+COMMAND_PATH_HELP = 'Path for generating cache'
```
Comment (ashedesimone) [inline edkrepo/commands/arguments/cache_args.py:20]: This should be 'Path where cache will be created' or similar

**Review by ashedesimone — CHANGES_REQUESTED** [review]
In addition to the comments about the code your commit message must be updated to include a description of the change and include your signed off by.

**Comment by ashedesimone** [comment]
> Updated Cache command to accept an optional path argument Signed-off-by: Harsh Vora [harsh.vora@intel.com](mailto:harsh.vora@intel.com)

Hi Harsh, updating the comment on the conversation page will not update the commit message. You will need to update the commit message in your local branch using git commit --amend and then force push an update to your PR branch

**Review by chrisyly — CHANGES_REQUESTED** [review]
We can improve the behavior enabling cache path. Expected behavior:
- when cache is enabled, and path is being given when enabling cache, storing the path in the edkrepo config.
- If path is being defined in edkrepo config, set cache to use the user defined cache path instead of default path

**Comment by ashedesimone** [comment]
> We can improve the behavior enabling cache path. Expected behavior:
> 
> * when cache is enabled, and path is being given when enabling cache, storing the path in the edkrepo config.
> * If path is being defined in edkrepo config, set cache to use the user defined cache path instead of default path

This is implemented in the most recent update to this PR

File: edkrepo/commands/cache_command.py [hunk 2]
```diff
@@ -57,18 +64,27 @@ def get_metadata(self):
         return metadata
 
     def run_command(self, args, config):
+        #print(dir(config['user_cfg_file']))
+        #raise
```
Comment (kevinsun49) [inline edkrepo/commands/cache_command.py:68]: Remove these commented out debug print statements

**Review by kevinsun49 — CHANGES_REQUESTED** [review]

File: edkrepo/commands/cache_command.py (hunk 2 above)
Comment (chrisyly) [reply edkrepo/commands/cache_command.py:68]: +1

**Review by chrisyly — APPROVED** [review]

**Review by kevinsun49 — APPROVED** [review]

**Pull request merged**